src/main/java/edu/hm/hafner/coverage/parser/GoParser.java

```diff
@@ -0,0 +1,277 @@
+        var filePaths = paths.subList(relativeImportOffset, paths.size());
+        //last element in the path is the file name
+        var fileName = paths.get(paths.size() - 1);
+        var filePath = String.join("/", filePaths);
+        //removing file from the paths as it is processed
+        FileNode fNode = meta.getOrAddFile(fileName, filePath);
+        addChildHelper(pNode, fNode);
+        return fNode;
+    }
+
+    private PackageNode initializePackageNode(final ModuleNode root, final List<String> paths, final GoCoverageMeta meta,
+                                              final int relativeImportOffset) {
+        //Removing the fist and last element as those are module and file
+        var packagePaths = paths.subList(relativeImportOffset, paths.size() - 1);
+        String packageName = String.join("/", packagePaths);
+        PackageNode pNode = meta.getOrAddPkg(packageName);
+        addChildHelper(root, pNode);
+        return pNode;
+    }
+
+    /**
+     * Initialize module node based on if is relative import of not.
+     *
+     * @param domainName takes the domain name
+     * @return if it relative import or not
+     */
+    private int fetchRelativePathOffset(final String domainName) {
+        //Format is:
+        //domain_name/root_dir/repo/dir_path.../file_name:<start-line>.<start-column>,<end-line>.<end-column> <statement-count>  <execution-count>
+        //where domain_name/root_dir is optional (i.e. relative imports)
+        return domainName.contains(".") ? 3 : 1; // reverse offset for relative imports
+    }
+
+    private void calculateCoverage(final FileNode fNode, final String coverageData) {
+        String[] split = coverageData.split("[, ]");
+        String[] start = split[0].split("\\.");
+        String startLine = start[0];
+        String[] end = split[1].split("\\.");
+        String endLine = end[0];
+        int statementCount = Integer.parseInt(split[split.length - 2]);
+        int executionCount = Integer.parseInt(split[split.length - 1]);
+        int j = Integer.parseInt(endLine);
+        for (int i = Integer.parseInt(startLine); i <= j; i++) {
+            if (executionCount > 0) {
+                fNode.addCounterIncremental(i, statementCount, 0);
+            }
+            else {
+                fNode.addCounterIncremental(i, 0, statementCount);
+            }
+        }
+    }
+
+    private void setCoverageMode(final String line, final FilteredLog log) throws InvalidGoCoverageFormatException {
+        String processedLine;
+        if (line.startsWith(GO_TEST_MODE)) {
+            processedLine = line.substring(GO_TEST_MODE.length());
+        }
+        else {
+            LOGGER.severe(USAGE_ERROR_MSG);
+            log.logError(USAGE_ERROR_MSG);
+            throw new InvalidGoCoverageFormatException("Expected the coverage mode specification, but got: [" + line + "]");
+        }
+        //coverage mode
+        CoverMode coverMode;
```

Comment:
The cover mode is not used at all?

Reply:
Makes sense. I can review this and update it.